File: LeetCode/387/387.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        array<int, 26> helper;
        std::fill(helper.begin(),helper.end(), 0);
        for(const auto& elem : s)
        {
            helper[elem - 'a'] +=1;
        }
        auto view = s | std::views::enumerate | std::views::filter([&](const auto& par)
        {
            auto [index, elem] = par;
            return helper[elem -'a'] == 1;
        });

        std::vector<pair<int,int>> vec(view.begin(), view.end());
        return vec.empty() ? -1 : vec[0].first;
    }
};

File: LeetCode/387/387.rs
impl Solution {
    pub fn first_uniq_char(s: String) -> i32 {
        let mut hash = vec![0i32; 26];
        for ch in s.chars()
        {
            hash[(ch as u8 - b'a') as usize] += 1;
        }
        s.chars().enumerate().position(|(index, chr)| hash[(chr as u8 - b'a') as usize] == 1).map(|index| index as i32)
        .unwrap_or(-1)
    }
}
